src/engine/renderer/tr_image.cpp

```diff
@@ -1361,6 +1361,24 @@ void R_UploadImage( const byte **dataArray, int numLayers, int numMips,
 		ri.Hunk_FreeTempMemory( scaledBuffer );
 	}
 
+	// detect heightmap in normalmap alpha channel
+	// and enable it
+	if ( image->bits & IF_NORMALMAP )
+	{
+		switch ( image->internalFormat )
+		{
+			case GL_RGBA:
+			case GL_RGBA8:
+			case GL_RGBA16:
+			case GL_RGBA16F:
+			case GL_RGBA32F:
+			case GL_RGBA32UI:
+			case GL_COMPRESSED_RGBA_S3TC_DXT3_EXT:
+			case GL_COMPRESSED_RGBA_S3TC_DXT5_EXT:
+				image->bits |= IF_DISPLACEMAP;
```

Comment:
This seems to happen too late as it already decides at [line 1018](https://github.com/DaemonEngine/Daemon/blob/d90cff38264a3e2b08383c0133ea5b833d027110/src/engine/renderer/tr_image.cpp#L1018) whether to have an alpha channel.

Reply:
I first thought about this, but if you look more precisely you'lle see that code is run when an `else` is reached:
https://github.com/DaemonEngine/Daemon/blob/d90cff38264a3e2b08383c0133ea5b833d027110/src/engine/renderer/tr_image.cpp#L1003-L1018

And for what I understand, this `else` is reached when the picture uses none of those formats.

I don't know for what is else block is for, but it looks to be there for when the image is not loaded from file, perhaps for when it's built from memory (and in this way the developer is able to set the `parallax` boolean to true).

Reply:
and it works :stuck_out_tongue:

Reply:
it did not worked :stuck_out_tongue: see #233

The fact that code deciding wheter to have alpha channel or not was broken because normalmap was excluded from automatic detection because of bad heightmap design.